ISX/ISX/RequestDetailViewController.swift

```diff
@@ -0,0 +1,144 @@
+        }else if(indexPath.row == (numberOfCells + (listWithFavorites.count - 1))) {
+            return Constants.requestExtraProductInfoCellSize
+        }else {
+            return Constants.requestLabelCellSize
+        }
+    }
+    
+    //Create each of the different cells, based on indexPath.
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        if(indexPath.row == 0) {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestProductInfo, for: indexPath) as? RequestProductInfoCell {
+                //Check later if all this data can be stored in one object.
+                cell.setCellData(productName: (self.productDetail?.title)!, usersChairNumber: chairNumber!, productNumber: (self.productDetail?.prologisticaNumberHH)!, productReference: self.requestReference!, isActive: self.request!.completed)
+                return cell
+            }
+        }else if(indexPath.row == 1) {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestLocationLabelCell, for: indexPath) as? RequestLocationLabelCell {
+                return cell
+            }
+        }else if(indexPath.row == 2) {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestLocationInfoCell, for: indexPath) as? RequestLocationInfoCell {
+                cell.setCellData(product: productDetail!)
+                return cell
+            }
+        }else if(indexPath.row == 3) {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestFavoritesLabelCell, for: indexPath) as? RequestFavoritesLabelCell {
+                return cell
+            }
+        }else if(indexPath.row >= 4 && indexPath.row < ((numberOfCells + (listWithFavorites.count) - 2))) {
+                if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.favoritesInRequestCell, for: indexPath) as? FavoritesInRequestCell {
+                    //indexPath.row - 4, because when we reach this cell, the index is already at 4, otherwise the array from favorites goes out of bounds.
+                    cell.setProductName(productName: listWithFavorites[(indexPath.row - 4)].title)
+                    return cell
+                }
+        }else if(indexPath.row == ((numberOfCells + listWithFavorites.count) - 1) ) {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestExtraProductDetailCell, for: indexPath) as? RequestExtraProductDetailCell {
+                return cell
+            }
+        }else {
+            if let cell = tableView.dequeueReusableCell(withIdentifier: Constants.requestExtraProductLabelCell, for: indexPath) as? RequestExtraProductDetailLabelCell {
+                return cell
+            }
+        }
+        return UITableViewCell()
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
```

Comment:
if it's empty then it can be removed

Reply:
heb uitgecomment, wordt gebruikt voor hotfix